File: frontend/types/Cluster.ts
import { Event } from "./Event";

export interface Cluster {
  id: number;
  subType: string;
  centerLatitude: number;
  centerLongitude: number;
  isResolved: boolean;
  eventCount: number;
  createdAt: string;
  latestEvent: Event | null;
  events?: Event[];
  neighborhood?: string | null;
}

File: backend/UrbanPulse.Core/Entities/Cluster.cs
namespace UrbanPulse.Core.Entities;

public class Cluster : BaseEntity
{
    public string SubType { get; set; } = string.Empty;
    public double CenterLatitude { get; set; }
    public double CenterLongitude { get; set; }
    public bool IsResolved { get; set; } = false;
    public string? Neighborhood { get; set; }
    public ICollection<Event> Events { get; set; } = new List<Event>();
}
